### PR TITLE
Refractor Search to TS

### PR DESCRIPTION
**Pull request by igbopie**
Refractor some files to TS and cleaned up some code.

### CODE REVIEW

File: wherehows-web/app/controllers/search.ts [hunk 1]
```diff
@@ -41,11 +85,13 @@ export default class Search extends Controller.extend({
...
-    var model = this.get('model');
+  }
+
+  @computed('model.page')
+  get first(): boolean {
+    const model = this.get('model');
     if (model && model.page) {
-      var currentPage = model.page;
+      const currentPage = model.page;
       if (currentPage <= 1) {
```
Comment (cptran777) [inline wherehows-web/app/controllers/search.ts]: Couldn't this just be return currentPage <= 1? Why all these weird if blocks taking up space?

File: wherehows-web/app/routes/search.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+   * Applies the returned results object as the route model and sets
+   *   keyword property on the route controller
+   * @param {Object} controller search route controller
+   * @param {Object} model search results
+   */
+  setupController(controller: SearchController, model: any) {
+    const { keywords } = model;
+
+    controller.setProperties({
+      model,
```
Comment (cptran777) [inline wherehows-web/app/routes/search.ts]: Confused by why we're setting model to model here on the controller. Doesn't that happen already by default?

Comment (igbopie) [reply wherehows-web/app/routes/search.ts]: That... was there haha

File: wherehows-web/app/controllers/search.ts (hunk 1 above)
Comment (igbopie) [reply wherehows-web/app/controllers/search.ts]: That... was there too!

Comment (igbopie) [reply wherehows-web/app/controllers/search.ts]: I am going to leave it for now. We will come back to this later.

File: wherehows-web/app/routes/search.ts (hunk 2 above)
Comment (igbopie) [reply wherehows-web/app/routes/search.ts]: Same, will try to refractor in another PR.

**Pull request merged**